util/network-devp2p/src/connection.rs

```diff
@@ -385,14 +388,18 @@ impl EncryptedConnection {
 		}
 		EncryptedConnection::update_mac(&mut self.ingress_mac, &mut self.mac_encoder, &header[0..16]);
 		let mac = &header[16..];
-		let mut expected = H256::new();
-		self.ingress_mac.clone().finalize(&mut expected);
+		let mut expected = H256::default();
+		self.ingress_mac.clone().finalize(expected.as_bytes_mut());
 		if mac != &expected[0..16] {
 			return Err(ErrorKind::Auth.into());
 		}
 
-		let mut hdec = H128::new();
-		self.decoder.decrypt(&mut RefReadBuffer::new(&header[0..16]), &mut RefWriteBuffer::new(&mut hdec), false).expect("Invalid length or padding");
+		let mut hdec = H128::default();
+		self.decoder.decrypt(
+			&mut RefReadBuffer::new(&header[0..16]),
+			&mut RefWriteBuffer::new(hdec.as_bytes_mut()),
+			false,
+		).expect("Invalid length or padding");
```

Comment:
We should not panic here. Not new code ofc, but unless we can prove this doesn't fail we need to fix this. Separate PR maybe?

Reply:
Add a todo + ref to https://github.com/paritytech/parity-ethereum/issues/10698?